lib/candy_check/app_store/verification.rb

```diff
@@ -37,8 +39,12 @@ def call!
 
       private
 
+      def response_status_ok?
+        @response['status'] == STATUS_OK
+      end
+
       def valid?
-        @response && @response['status'] == STATUS_OK && @response['receipt']
+        @response && response_status_ok? && @response['receipt']
```

Comment:
It feels a bit like `response_status_ok?` is not self-contained if you have to check `@response` before you use it. Maybe it would be nicer with two black boxes: `response_status_ok?` and `response_has_receipt?` which both take care of validating all that they need. This would mean duplicating the `@response` check, but I think that's acceptable, or if the response is only set once, then you could even cache/memoize this in a `has_response?` method